dart/dynamics/BodyNode.h

```diff
@@ -613,6 +591,60 @@ class BodyNode :
   template <class NodeType, typename ...Args>
   NodeType* createNode(Args&&... args);
 
+  /// Create an ShapeNode attached to this BodyNode. Pass an
+  /// ShapeNode::Properties argument into this function.
+  template <class ShapeNodeProperties>
+  ShapeNode* createShapeNode(const ShapeNodeProperties& properties);
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(
+      const ShapePtr& shape, const std::string& name = "ShapeNode");
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(const ShapePtr& shape, const char* name);
+
+  /// Create an ShapeNode with the specified name and addons
+  template <class... Addons>
+  ShapeNode* createShapeNode(const ShapePtr& shape,
+                             const std::string& name = "ShapeNode");
```

Comment:
This function was added exactly for the case you mentioned. One might want to create a `ShapeNode` with several addons as soon as it gets created. [Here is one example case](https://github.com/dartsim/dart/blob/shapenode/tutorials/tutorialBiped-Finished.cpp#L439). Without this function, we need multiple lines for this like:
```cpp
auto shapeNode = bodyNode->createShapeNode(shape);
shapeNode->createVisualAddon();
shapeNode->createCollisionAddon();
shapeNode->createDynamicsAddon();
```
I don't object to remove this function, but this would be useful for the above use case.